bundles/org.openhab.binding.boschspexor/src/main/java/org/openhab/binding/boschspexor/internal/BoschSpexorThingHandler.java

```diff
@@ -0,0 +1,369 @@
+            }
+            result = spexor;
+        }
+        return result;
+    }
+
+    public void discoverChannels() {
+        SpexorInfo spexor = cache.getValue();
+        if (spexor != null) {
+            boolean thingStructureChanged = false;
+            ThingBuilder thingBuilder = editThing();
+
+            for (ObservationStatus observationStatus : spexor.getStatus().getObservation()) {
+                String observationType = observationStatus.getObservationType();
+                Channel channel = getThing().getChannel(getChannelID(GROUP_ID_OBSERVATIONS, observationType));
+                if (channel == null) {
+                    channel = createObservationChannel(observationType);
+                    thingBuilder.withoutChannel(channel.getUID()).withChannel(channel);
+                    thingStructureChanged = true;
+                }
+                updateState(channel.getUID(), new StringType(observationStatus.getSensorMode().name()));
+            }
+
+            // SENSORS
+            for (String sensor : spexor.getSensors()) {
+                String sensorType = sensor;
+                Channel channel = getThing().getChannel(getChannelID(GROUP_ID_SENSORS, sensorType));
+                if (channel == null) {
+                    channel = createSensorChannel(sensor, sensorType);
+                    if (channel != null) {
+                        thingBuilder.withoutChannel(channel.getUID()).withChannel(channel);
+                        thingStructureChanged = true;
+                    }
+                }
+
+            }
+
+            if (thingStructureChanged) {
+                updateThing(thingBuilder.build());
+                if (logger.isDebugEnabled()) {
+                    logger.debug("structural change of thing \"{}\"", getThing().getUID());
+                }
+            }
+        }
+    }
+
+    private Channel createObservationChannel(String observationType) {
+        Channel channel = ChannelBuilder.create(getChannelID(GROUP_ID_OBSERVATIONS, observationType), "String")
+                .withType(new ChannelTypeUID("boschspexor", "sampleObservationState"))
+                .withLabel("Observation Type " + observationType).build();
+        return channel;
+    }
+
+    private @Nullable Channel createSensorChannel(String sensor, String sensorType) {
+        Channel channel = null;
+        switch (sensor) {
+            case TYPE_AIR_QUALITY:
+            case TYPE_TEMPERATURE:
+            case TYPE_HUMIDITY:
+            case TYPE_MICROPHONE:
+            case TYPE_FIRE:
+            case TYPE_PRESSURE:
+            case TYPE_ACCELERATION:
+            case TYPE_LIGHT:
+            case TYPE_GAS:
+            case TYPE_PASSIVE_INFRARED:
+                channel = ChannelBuilder.create(getChannelID(GROUP_ID_SENSORS, sensorType), "Number")
+                        .withType(new ChannelTypeUID("boschspexor", "sampleNumberChannel")).withKind(ChannelKind.STATE)
```

Comment:
Suggest to remove redundant information. By specifying the type this information is already defined in the type itself and therefor doesn't need to specified explicitly: 
```suggestion
                channel = ChannelBuilder.create(getChannelID(GROUP_ID_SENSORS, sensorType))
                        .withType(new ChannelTypeUID("boschspexor", "sampleNumberChannel"))
```

You could consider using QuantityType channel types for these channels instead of plain Number channels. It's a bit more work because you need to specify the specific type per channel. But will give the user more control over the way to display the values and use them as QuantityType channels.